File: mrs/apps/genblaze-media/scripts/flux_smoke.py
#!/usr/bin/env python3
"""Direct NVIDIA FLUX.1-schnell smoke (Python httpx). Never prints the API key."""

from __future__ import annotations

import os
import sys
from pathlib import Path

import httpx
from dotenv import load_dotenv

ROOT = Path(__file__).resolve().parents[4]
load_dotenv(ROOT / ".env", override=True)

KEY = (
    os.getenv("NVIDIA_API_KEY")
    or os.getenv("NVIDIA_NIM_API_KEY")
    or os.getenv("NGC_API_KEY")
    or ""
).strip()
if not KEY:
    print("NVIDIA_API_KEY missing", file=sys.stderr)
    sys.exit(1)

URL = "https://ai.api.nvidia.com/v1/genai/black-forest-labs/flux.1-schnell"
prompt = sys.argv[1] if len(sys.argv) > 1 else "a simple coffee shop interior"
payload = {
    "prompt": prompt,
    "width": 512,
    "height": 512,
    "seed": 0,  # NVIDIA sample leaves seed empty — that is invalid JSON
    "steps": 4,
}

try:
    r = httpx.post(
        URL,
        headers={
            "Authorization": f"Bearer {KEY}",
            "Accept": "application/json",
            "Content-Type": "application/json",
        },
        json=payload,
        timeout=600.0,
    )
except httpx.TimeoutException as exc:
    print(f"flux_timeout {exc}", file=sys.stderr)
    sys.exit(1)

print("http", r.status_code)
if not r.is_success:
    print(r.text[:400], file=sys.stderr)
    sys.exit(1)

body = r.json()
print("flux_ok keys", ",".join(body.keys()))
img = body.get("image")
if isinstance(img, str):
    print("image_b64_len", len(img))

File: mrs/apps/genblaze-media/scripts/flux-smoke.mjs
#!/usr/bin/env node
/**
 * Direct NVIDIA FLUX.1-schnell smoke (Node).
 * Loads repo-root .env — never prints the key.
 *
 * Usage (from repo root):
 *   node mrs/apps/genblaze-media/scripts/flux-smoke.mjs
 */
import { readFileSync, existsSync } from "node:fs";
import { resolve, dirname } from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = dirname(fileURLToPath(import.meta.url));
const repoRoot = resolve(__dirname, "../../../..");
const envPath = resolve(repoRoot, ".env");

function loadEnv(path) {
  if (!existsSync(path)) return;
  for (const line of readFileSync(path, "utf8").split(/\r?\n/)) {
    const t = line.trim();
    if (!t || t.startsWith("#")) continue;
    const i = t.indexOf("=");
    if (i < 0) continue;
    const k = t.slice(0, i).trim();
    let v = t.slice(i + 1).trim();
    if (
      (v.startsWith('"') && v.endsWith('"')) ||
      (v.startsWith("'") && v.endsWith("'"))
    ) {
      v = v.slice(1, -1);
    }
    if (!(k in process.env)) process.env[k] = v;
  }
}

loadEnv(envPath);

const key =
  process.env.NVIDIA_API_KEY ||
  process.env.NVIDIA_NIM_API_KEY ||
  process.env.NGC_API_KEY;
if (!key) {
  console.error("NVIDIA_API_KEY missing in .env");
  process.exit(1);
}

const invokeUrl =
  "https://ai.api.nvidia.com/v1/genai/black-forest-labs/flux.1-schnell";
const payload = {
  prompt: process.argv[2] || "a simple coffee shop interior",
  width: 512,
  height: 512,
  seed: 0,
  steps: 4,
};

const controller = new AbortController();
const timer = setTimeout(() => controller.abort(), 180_000);

try {
  const response = await fetch(invokeUrl, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${key}`,
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
    signal: controller.signal,
  });
  clearTimeout(timer);
  const text = await response.text();
  if (!response.ok) {
    console.error("flux_fail", response.status, text.slice(0, 400));
    process.exit(1);
  }
  const body = JSON.parse(text);
  const keys = Object.keys(body);
  console.log("flux_ok", response.status, "keys", keys.join(","));
  if (typeof body.image === "string") {
    console.log("image_b64_len", body.image.length);
  }
} catch (err) {
  clearTimeout(timer);
  console.error("flux_error", err?.name || err, String(err?.message || err).slice(0, 200));
  process.exit(1);
}
